src/highlight_manager.ts

```diff
@@ -196,7 +196,7 @@ export class HighlightManager implements Disposable, NeovimRedrawProcessable, Ne
                         // use a single text decoration but modify it when a
                         // new decoration would be pushed on top of it.
                         const ogText = drawnAt.get(mapKey).renderOptions.after.contentText;
-                        drawnAt.get(mapKey).renderOptions.after.contentText = (text[0] + ogText).substr(
+                        drawnAt.get(mapKey).renderOptions.after.contentText = (text[0] + ogText).substring(
                             0,
                             ogText.length,
```

Comment:
The 2nd param of substring() is "end" , not "length". Are you sure this is correct?

Reply:
In this case, yes, they act the same because the start index is zero.

But you're right, in the case where the start index is non-zero, they act differently.

![image](https://user-images.githubusercontent.com/51082452/219024148-523f8a76-a39e-4120-9bf8-a1cff8934b4c.png)

Reply:
should we drop the `ogText.length` argument here?